graph/graph.go

```diff
@@ -85,6 +85,12 @@ func (g *Graph) Get(id string) interface{} {
 
 // GetParent returns the direct parent vertex of the current node.
 func (g *Graph) GetParent(id string) interface{} {
+	parentID := g.GetParentID(id)
+	return g.Get(parentID)
+}
+
+// GetParentID uses edge traversal to lookup the parent of the current node.
+func (g *Graph) GetParentID(id string) string {
```

Comment:
it'll make it easier for our changes (mine in #369) to merge if this is `(id string, ok bool)`. Explicitly identifies the edge condition, as well.